### PR TITLE
feat(workspace): add datasource to get the list of desktops

### PR DESCRIPTION
**Pull request by wuzhuanhong**
**What this PR does / why we need it**:

Add workspace desktops datasource.

New dataSource:
+ huaweicloud_workspace_desktops

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. using this data source to get the list of Workspace desktops.
2. support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDataSourceDesktops_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDataSourceDesktops_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDesktops_basic
--- PASS: TestAccDataSourceDesktops_basic (524.77s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 524.813s
```

### CODE REVIEW

File: docs/data-sources/workspace_desktops.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `desktop_id` - (Optional, String) Specifies ID of the desktop.
+
+* `name` - (Optional, String) Specifies the desktop name of the desktop.
+
+* `user_name` - (Optional, String) Specifies the user name to which the desktops belongs .
```
Comment (Lance52259) [inline docs/data-sources/workspace_desktops.md]:
```suggestion
* `user_name` - (Optional, String) Specifies the user name to which the desktops belongs.
```

File: docs/data-sources/workspace_desktops.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `desktop_id` - (Optional, String) Specifies ID of the desktop.
+
+* `name` - (Optional, String) Specifies the desktop name of the desktop.
```
Comment (Lance52259) [inline docs/data-sources/workspace_desktops.md]:
```suggestion
* `name` - (Optional, String) Specifies the desktop name.
```
or
```suggestion
* `name` - (Optional, String) Specifies the name of the desktop.
```

File: docs/data-sources/workspace_desktops.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+* `id` - The data source ID.
+
+* `desktops` - The list of the desktops.  
+  The [desktops](#workspace_desktops) structure is documented below.
+
+<a name="workspace_desktops"></a>
+The `desktops` block supports:
+
+* `desktop_id` - The desktop ID.
```
Comment (Lance52259) [inline docs/data-sources/workspace_desktops.md]:
```suggestion
* `id` - The desktop ID.
```
please check the schema definition.

File: huaweicloud/provider.go [hunk 4]
```diff
@@ -658,7 +658,8 @@ func Provider() *schema.Provider {
 			"huaweicloud_waf_address_groups":      waf.DataSourceWafAddressGroups(),
 			"huaweicloud_dws_flavors":             dws.DataSourceDwsFlavors(),
 
-			"huaweicloud_workspace_flavors": workspace.DataSourceWorkspaceFlavors(),
+			"huaweicloud_workspace_flavors":  workspace.DataSourceWorkspaceFlavors(),
+			"huaweicloud_workspace_desktops": workspace.DataSourceDesktops(),
```
Comment (Lance52259) [inline huaweicloud/provider.go]:
```suggestion
			"huaweicloud_workspace_desktops": workspace.DataSourceDesktops(),
			"huaweicloud_workspace_flavors":  workspace.DataSourceWorkspaceFlavors(),
```
reorder please

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDataSourceDesktops_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDataSourceDesktops_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceDesktops_basic
--- PASS: TestAccDataSourceDesktops_basic (564.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 564.935s

File: huaweicloud/provider.go (hunk 4 above)
Comment (wuzhuanhong) [reply huaweicloud/provider.go]: fixed

File: docs/data-sources/workspace_desktops.md (hunk 3 above)
Comment (wuzhuanhong) [reply docs/data-sources/workspace_desktops.md]: fixed

File: docs/data-sources/workspace_desktops.md (hunk 1 above)
Comment (wuzhuanhong) [reply docs/data-sources/workspace_desktops.md]: fixed

File: docs/data-sources/workspace_desktops.md (hunk 2 above)
Comment (wuzhuanhong) [reply docs/data-sources/workspace_desktops.md]: fixed

File: docs/data-sources/workspace_desktops.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+<a name="desktop_attach_user_infos"></a>
+The `attach_user_infos` block supports:
+
+* `user_group` - The user group to which the desktop belongs.
+
+* `user_name` - The user name to which the desktop belongs.
+
+<a name="desktop_volume"></a>
+The `data_volume` and `data_volume` block supports:
```
Comment (Lance52259) [inline docs/data-sources/workspace_desktops.md]:
```suggestion
The `root_volume` and `data_volume` block supports:
```

Comment (wuzhuanhong) [reply docs/data-sources/workspace_desktops.md]: fixed

**Comment by wuzhuanhong** [comment]
make testacc TEST=./huaweicloud/services/acceptance/workspace TESTARGS='-run TestAccDataSourceDesktops_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/workspace -v -run TestAccDataSourceDesktops_basic -timeout 360m -parallel 4
=== RUN TestAccDataSourceDesktops_basic
--- PASS: TestAccDataSourceDesktops_basic (564.87s)
PASS
ok github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/workspace 564.935s

**Pull request merged**